Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewDataErrorEventArgs.cs

```diff
@@ -0,0 +1,68 @@
+
+using System;
+using System.Diagnostics.Contracts;
+
+namespace System.Windows.Forms
+{
+    /// <summary>
+    /// Provides data for the <see cref="E:System.Windows.Forms.DataGridView.DataError"/> event.
+    /// </summary>
+    
+    public class DataGridViewDataErrorEventArgs : DataGridViewCellCancelEventArgs
+    {
+
+        /// <summary>
+        /// Gets details about the state of the <see cref="T:System.Windows.Forms.DataGridView"/> when the error occurred.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// A bitwise combination of the <see cref="T:System.Windows.Forms.DataGridViewDataErrorContexts"/> values that specifies the context in which the error occurred.
+        /// </returns>
+        /// 
+        // public DataGridViewDataErrorContexts Context {get;}
+
+        /// <summary>
+        /// Gets the exception that represents the error.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// An <see cref="T:System.Exception"/> that represents the error.
+        /// </returns>
+        /// 
+        // public Exception Exception {get;}
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to throw the exception after the <see cref="T:System.Windows.Forms.DataGridViewDataErrorEventHandler"/> delegate is finished with it.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// true if the exception should be thrown; otherwise, false. The default is false.
+        /// </returns>
+        /// <exception cref="T:System.ArgumentException">When setting this property to true, the <see cref="P:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception"/> property value is null.</exception>
+        // public bool ThrowException { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:System.Windows.Forms.DataGridViewDataErrorEventArgs"/> class.
+        /// </summary>
+        /// <param name="exception">The exception that occurred.</param><param name="columnIndex">The column index of the cell that raised the <see cref="E:System.Windows.Forms.DataGridView.DataError"/>.</param><param name="rowIndex">The row index of the cell that raised the <see cref="E:System.Windows.Forms.DataGridView.DataError"/>.</param><param name="context">A bitwise combination of <see cref="T:System.Windows.Forms.DataGridViewDataErrorContexts"/> values indicating the context in which the error occurred. </param>
+        public DataGridViewDataErrorEventArgs(Exception exception, int columnIndex, int rowIndex,
+            DataGridViewDataErrorContexts context)
+          : base(columnIndex, rowIndex)
+        {
```

Comment:
They are on the base constructor, should I add them here as well. If not, I will need to check for redundant contracts in subclasses.